assets/scripts/segments/ResizeHandle.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { DragSource } from 'react-dnd'
+import { getEmptyImage } from 'react-dnd-html5-backend'
+import { DragTypes } from './constants'
+import { beginSegmentResize, endSegmentResize } from './resize_drag'
+import store from '../store'
+import { updateResizeDragState } from '../store/actions/ui'
+
+const dragSpec = {
+  beginDrag (props, monitor, component) {
+    store.dispatch(updateResizeDragState(true))
```

Comment:
I know both infoBubble and the dragGuides depend on the `resizeDragState` to show/hide but I was wondering if it would be better to check for the class 'segment-resize-dragging' under document.body instead to avoid another action dispatch? I noticed that if we didn't dispatch the `updateResizeDragState` it _looks_ like it's a slightly better performance (I didn't check document.body.classList though so maybe that will slow it down as well?)

Also speaking of improving performance, I was reading the thread on this issue https://github.com/react-dnd/react-dnd/issues/456
and one of the suggestions was to add `transform: translate3d(0,0,0)` to components with a complicated DOM tree and so I tested it with 'street-section-canvas' dom element which would get rerendered every time a segment is resized since segment positions change and once again it _looks_ slightly better?

Reply:
I don't think dispatch itself is the problem, it's what happens when anything reacts to the dispatch. Dispatching is just calling a function and updating state in JavaScript objects, both of which are super cheap. But every dispatch potentially triggers component re-renders. I think that's the part we have to be careful of. I'm more concerned about the changes that occur during the drag because that could happen many times a second and every dispatch causes DOM to change.

It's possible if we altered `document.body.classList` there's less of a problem. If we never add any styling to that class, the browser shouldn't repaint (which can be expensive). But I think using the body class list as state is a bad idea when we have a state object. We should only be using `document.body.classList` when we need the class for styling purposes.

As for the `transform: translate3d(0,0,0)` trick, this can tend to work because this automatically tells the browser to use the GPU to process layout, instead of the CPU. But we should profile it and see how much better we're getting. I think we can get slight gains with it but I think the bulk of the gains will come from optimizing the dispatch and re-renders that happen during the resize action itself.